### PR TITLE
- Adding support for launching dialogs from the grade item buttons

### PR DESCRIPTION
**Pull request by jmurray-d2l**
- Looking for feedback

### CODE REVIEW

File: src/components/d2l-grade-result-presentational.js [hunk 1]
```diff
@@ -162,6 +166,80 @@ export class D2LGradeResultPresentational extends LocalizeMixin(LitElement) {
...
 
+	_openGradeEvaluationDialog() {
+
+		const dialogUrl = this.gradeButtonUrl
+
+		if (!dialogUrl) {
+			return;
+		}
+
+		const location = new D2L.LP.Web.Http.UrlLocation(dialogUrl);
```
Comment (AlexBedley) [inline src/components/d2l-grade-result-presentational.js:177]: IMO the dialog launching does not belong here. I'm loath to tie the LMS to this presentational component. I might be convinced if every single grade-result ever would launch the exact same dialog etc. But even in that case this code shouldn't be in the presentational layer (I'd write a wrapper around this class that has dialogs).

My thought is we just listen for the 2 button presses (`d2l-grade-result-grade-button-click` and `d2l-grade-result-reports-button-click`) in your other PR, and launch the dialogs from there.

Comment (jmurray-d2l) [reply src/components/d2l-grade-result-presentational.js:177]: Cool yeah, I would expect the same dialog to be launched each time, but I wasn't aware that I could hook into this component from the other, that seems like a better idea - thanks

**Comment by jmurray-d2l** [comment]
In going with the suggestion Alex gave, this PR is no longer needed

File: src/components/d2l-grade-result-presentational.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ export class D2LGradeResultPresentational extends LocalizeMixin(LitElement) {
 			scoreNumerator: { type: Number },
 			letterGradeOptions: { type: Array },
 			selectedLetterGrade: { type: String },
-			includeGradeButton: { type: Boolean },
-			includeReportsButton: { type: Boolean },
+			gradeButtonUrl: { type: String },
+			reportsButtonUrl: { type: String },
```
Comment (zhangium) [inline src/components/d2l-grade-result-presentational.js:20]: don't you still need these changes? everywhere that `includeGradeButton` and `includeReportsButton` bools get replaced by the urls

**Pull request closed without merging**